guacamole/src/main/java/org/apache/guacamole/rest/connection/ConnectionResource.java

```diff
@@ -119,11 +120,13 @@ public ConnectionResource(@Assisted UserContext userContext,
     public Map<String, String> getConnectionParameters()
             throws GuacamoleException {
 
+        // Pull effective permissions
         User self = userContext.self();
+        Permissions effective = self.getEffectivePermissions();
```

Comment:
In the `ConnectionGroupTree` class you did:

    Permissions effective = userContext.self().getEffectivePermissions();

eliminating the need for the existence of the `self` object.  Any reason not to do the same, here?